py4web/utils/grid.py

```diff
@@ -444,7 +444,9 @@ def process(self):
             # for all FieldVirtual or Custom columns, add all columns from their table
             # as they might be needed for processing
             for col in [
-                x for x in self.param.columns if isinstance(x, (FieldVirtual, Column))
+                x
+                for x in self.param.columns
+                if isinstance(x, (FieldVirtual, Column)) and "tablename" in dir(x)
```

Comment:
I do not think a Column can have a tablename attribute and if it did it may be intended for a different purpose.
I think we should just do:

    for col in self.param.columns
        if isinstance(col, VirtualField):
           ....

Reply:
I can test that tomorrow with the html_grid example as well as my other grid instances including virtual fields.